### ISSUE TITLE
Settings icons are color inverted based on wrong metrix

### ISSUE THREAD

**Issue by nickvergessen** [nextcloud/server#41352]
The icons of the `app-navigation-*` are influenced by the dark/bright theme. However for the active entry this behaviour is wrong and it would always need to use the color of `var(--color-primary-element-text)` instead of a filter `var(--background-invert-if-dark)`

Basically we need the opposite of https://github.com/nextcloud/server/blob/3378a73e99171ac31860ccca81b1b3197e63417f/apps/theming/lib/Themes/CommonThemeTrait.php#L55

Bright theme | Dark theme
---|---
![Bildschirmfoto vom 2023-11-09 07-18-08](https://github.com/nextcloud/server/assets/213943/6699283c-49d2-41f2-a6e4-f4076b07e6d2) | ![Bildschirmfoto vom 2023-11-09 07-18-04](https://github.com/nextcloud/server/assets/213943/862cd1bd-7644-4c56-85d6-42f2d51bd6df)
![Bildschirmfoto vom 2023-11-09 07-18-26](https://github.com/nextcloud/server/assets/213943/415bf456-e5e2-45a4-a734-c2d60cb784b5) | ![Bildschirmfoto vom 2023-11-09 07-18-30](https://github.com/nextcloud/server/assets/213943/53e658a5-320e-48da-b6f2-23cf9088f572)

**Comment by szaimen** [comment]
I think the problem is rather that the icons are loaded via url and cannot be inverted thus...

**Comment by nickvergessen** [comment]
> I think the problem is rather that the icons are loaded via url and cannot be inverted thus...

But they are inverted, just using the wrong "foundation" color to decide if it should be inverted. as per above I think we just need to add a new constant which does the invert of the existing variable.

**Comment by szaimen** [comment]
And even better would be using `--color-primary-element-text` for the icon but that does not work with icons that are loaded via url...

**Comment by nickvergessen** [comment]
I'll send the filter patch I mentioned above

**Comment by nickvergessen** [comment]
Patch at https://github.com/nextcloud/server/pull/41855

**Comment by nickvergessen** [comment]
> And even better would be using `--color-primary-element-text` for the icon but that does not work with icons that are loaded via url...

yes, but it would break the API so not an option atm

**Comment by szaimen** [comment]
Thanks Joas! :)
